lib/Wikia/src/Service/User/Preferences/PreferenceService.php

```diff
@@ -2,26 +2,243 @@
 
-use Wikia\Domain\User\GlobalPreference;
+use Doctrine\Common\Cache\CacheProvider;
+use Wikia\Domain\User\Preferences\LocalPreference;
+use Wikia\Domain\User\Preferences\UserPreferences;
+use Wikia\Logger\Loggable;
+use Wikia\Persistence\User\Preferences\PreferencePersistence;
+use Wikia\Util\WikiaProfiler;
 
-interface PreferenceService {
+class PreferenceService {
+
+	use WikiaProfiler;
+	use Loggable;
+
+	const CACHE_PROVIDER = "user_preferences_cache_provider";
+	const HIDDEN_PREFS = "user_preferences_hidden_prefs";
+	const DEFAULT_PREFERENCES = "user_preferences_default_prefs";
+	const FORCE_SAVE_PREFERENCES = "user_preferences_force_save_prefs";
+	const PROFILE_EVENT = \Transaction::EVENT_USER_PREFERENCES;
+
+	/** @var CacheProvider */
+	private $cache;
+
+	/** @var PreferencePersistence */
+	private $persistence;
+
+	/** @var UserPreferences[string] */
+	private $preferences;
+
+	/** @var string[] */
+	private $hiddenPrefs;
+
+	/** @var string[] */
+	private $defaultPreferences;
+
+	/** @var string[] */
+	private $forceSavePrefs;
+
+	/**
+	 * @Inject({
+	 *    Wikia\Service\User\Preferences\PreferenceService::CACHE_PROVIDER,
+	 *    Wikia\Persistence\User\Preferences\PreferencePersistence::class,
+	 *    Wikia\Service\User\Preferences\PreferenceService::HIDDEN_PREFS,
+	 *    Wikia\Service\User\Preferences\PreferenceService::DEFAULT_PREFERENCES,
+	 *    Wikia\Service\User\Preferences\PreferenceService::FORCE_SAVE_PREFERENCES})
+	 * @param CacheProvider $cache,
+	 * @param PreferencePersistence $persistence
+	 * @param string[] $hiddenPrefs
+	 * @param string[string] $defaultPrefs
+	 * @param string[] $forceSavePrefs
+	 */
+	public function __construct(
+		CacheProvider $cache,
+		PreferencePersistence $persistence,
+		$hiddenPrefs,
+		$defaultPrefs,
+		$forceSavePrefs) {
+
+		$this->cache = $cache;
+		$this->persistence = $persistence;
+		$this->hiddenPrefs = $hiddenPrefs;
+		$this->defaultPreferences = $defaultPrefs;
+		$this->forceSavePrefs = $forceSavePrefs;
+		$this->preferences = [];
+	}
+
+	public function getPreferences($userId) {
+		return $this->load($userId);
+	}
+
+	public function getGlobalPreference($userId, $name, $default = null, $ignoreHidden = false) {
+		if (in_array($name, $this->hiddenPrefs) && !$ignoreHidden) {
+			return $this->getFromDefault($name);
+		}
+
+		$preferences = $this->load($userId);
+		if ($preferences->hasGlobalPreference($name)) {
+			return $preferences->getGlobalPreference($name);
+		}
+
+		return $default;
+	}
+
+	public function setGlobalPreference($userId, $name, $value) {
+		if ($value == null) {
+			$value = $this->getFromDefault($name);
+		}
+
+		$this->load($userId)->setGlobalPreference($name, $value);
+	}
+
+	public function deleteGlobalPreference($userId, $name) {
+		$this->load($userId)->deleteGlobalPreference($name);
+	}
+
+	public function getLocalPreference($userId, $wikiId, $name, $default = null, $ignoreHidden = false) {
+		if (in_array($name, $this->hiddenPrefs) && !$ignoreHidden) {
```

Comment:
I think that's the intention, though I'm unsure. This is duplicated logic from what MW was doing.